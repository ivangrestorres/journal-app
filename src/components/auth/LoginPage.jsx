import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { startGoogleLogin, startLoginEmailPassword } from "../../actions/auth";
import { useForm } from "../../hooks/useForm";

export const LoginPage = () => {
    const dispatch = useDispatch();
    const { loading, msgLoginError } = useSelector((state) => state.ui);

    const initialForm = {
        email: "kk@gmail.com",
        password: "12345aA",
    };
    const [formValues, handleInputChange] = useForm(initialForm);

    const { email, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(startLoginEmailPassword(email, password));
    };

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    };

    return (
        <div>
            <h3 className={"auth_title"}>Login</h3>
            {msgLoginError && (
                <div className={"auth_alert-error"}>{msgLoginError}</div>
            )}
            <form onSubmit={handleLogin}>
                <input
                    type="text"
                    placeholder={"Email"}
                    name={"email"}
                    className={"auth_input"}
                    autoComplete={"off"}
                    value={email}
                    onChange={handleInputChange}
                />
                <input
                    type="password"
                    placeholder={"Password"}
                    name={"password"}
                    className={"auth_input"}
                    value={password}
                    onChange={handleInputChange}
                />
                <button
                    className={"btn btn-primary btn-block"}
                    type={"submit"}
                    disabled={loading}
                >
                    Login
                </button>

                <div className={"auth_social-networks"}>
                    <p>Login with social networks</p>
                    <div className="google-btn" onClick={handleGoogleLogin}>
                        <div className="google-icon-wrapper">
                            <img
                                className="google-icon"
                                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                                alt="google button"
                            />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>
                <Link className={"link"} to={"/auth/register"}>
                    Create new account
                </Link>
            </form>
        </div>
    );
};
