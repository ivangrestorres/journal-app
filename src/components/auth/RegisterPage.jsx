import React from "react";
import { Link } from "react-router-dom";

export const RegisterPage = () => {
    return (
        <div>
            <h3 className={"auth_title"}>Register</h3>
            <form action="">
                <input
                    type="text"
                    placeholder={"Name"}
                    name={"name"}
                    className={"auth_input"}
                    autoComplete={"off"}
                />
                <input
                    type="text"
                    placeholder={"Email"}
                    name={"email"}
                    className={"auth_input"}
                    autoComplete={"off"}
                />
                <input
                    type="password"
                    placeholder={"Password"}
                    name={"password"}
                    className={"auth_input"}
                />
                <input
                    type="password"
                    placeholder={"Confirm password"}
                    name={"password2"}
                    className={"auth_input"}
                />
                <button
                    className={"btn btn-primary btn-block mt-5 mb-5"}
                    type={"submit"}
                >
                    Register
                </button>
                <Link className={"link"} to={"/auth/login"}>
                    Already register?
                </Link>
            </form>
        </div>
    );
};
