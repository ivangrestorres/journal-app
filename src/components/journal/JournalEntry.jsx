import React from "react";

export const JournalEntry = () => {
    return (
        <div className={"journal_entry"}>
            <div
                className="journal_entry-picture"
                style={{
                    backgroundSize: "cover",
                    backgroundImage:
                        "url(https://st3.depositphotos.com/1813774/19042/i/600/depositphotos_190424116-stock-photo-historical-medieval-city-along-the.jpg)",
                }}
            ></div>
            <div className={"journal_entry-body"}>
                <p className={"journal_entry-title"}>Un nuevo dia</p>
                <p className={"journal_entry-content"}>
                    Est sint officia fugiat irure anim magna aliquip sit qui
                    mollit enim nisi occaecat anim. Ut duis aute mollit culpa
                    deserunt eiusmod incididunt aute cillum officia non. Aute
                    nisi aute anim in dolor et anim eu. Officia duis dolore
                    pariatur ut elit cupidatat dolor nostrud.
                </p>
            </div>
            <div className={"journal_entry-date-box"}>
                <span>Monday</span>
                <h4>28</h4>
            </div>
        </div>
    );
};
