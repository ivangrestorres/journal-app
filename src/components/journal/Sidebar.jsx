import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startLogout } from "../../actions/auth";
import { addNote } from "../../actions/notes";
import { JournalEntries } from "./JournalEntries";

export const Sidebar = () => {
    const dispatch = useDispatch();
    const { name } = useSelector((state) => state.auth);

    const handleLogOut = () => {
        dispatch(startLogout());
    };

    const handleAddNewNote = () => {
        dispatch(addNote());
    };

    return (
        <aside className={"journal_sidebar"}>
            <div className={"journal_sidebar-navbar"}>
                <h4 className={"mt-1"}>
                    <i className={"far fa-user"}></i>
                    <span> {name}</span>
                </h4>
                <button onClick={handleLogOut} className={"btn"}>
                    Logout
                </button>
            </div>
            <div onClick={handleAddNewNote} className={"journal_new-entry"}>
                <i className="far fa-calendar-plus fa-3x"></i>
                <p className="mt-5">New entry</p>
            </div>
            <JournalEntries />
        </aside>
    );
};
