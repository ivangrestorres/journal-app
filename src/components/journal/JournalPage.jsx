import React from "react";
import { NoteScreen } from "../notes/NoteScreen";
// import { NothingSelected } from "./NothingSelected";
import { Sidebar } from "./Sidebar";

export const JournalPage = () => {
    return (
        <div className={"journal_main-content"}>
            <Sidebar />
            <main>
                {/* <NothingSelected /> */}
                <NoteScreen />
            </main>
        </div>
    );
};
