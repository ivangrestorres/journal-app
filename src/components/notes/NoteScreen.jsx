import React from "react";
import { useRef } from "react";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { activeNote, startDeleting } from "../../actions/notes";
import { useForm } from "../../hooks/useForm";
import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
    const dispatch = useDispatch();

    const { active: note } = useSelector((state) => state.notes);

    const [formValues, handleInputChange, reset] = useForm(note);
    const { title, body } = formValues;

    const activeId = useRef(note.id);

    useEffect(() => {
        if (note.id !== activeId.current) {
            reset(note);
            activeId.current = note.id;
        }
    }, [note, reset]);

    useEffect(() => {
        dispatch(activeNote(formValues.id, { ...formValues }));
    }, [formValues, dispatch]);

    const handleDelete = () => {
        dispatch(startDeleting(note.id));
    };

    return (
        <div className={"notes_main-content"}>
            <NotesAppBar />
            <div className={"notes_content"}>
                <input
                    type="text"
                    placeholder={"Some awesome title"}
                    className={"notes_title-input"}
                    name={"title"}
                    value={title}
                    onChange={handleInputChange}
                    autoComplete={"off"}
                />
                <textarea
                    placeholder={"What happened today"}
                    value={body}
                    name={"body"}
                    onChange={handleInputChange}
                    className={"notes_textarea"}
                ></textarea>
                {note.url && (
                    <div>
                        <img
                            src={note.url}
                            alt="imagen"
                            className={"notes_image"}
                        />
                    </div>
                )}
            </div>
            <button onClick={handleDelete} className={"btn btn-danger"}>
                Delete
            </button>
        </div>
    );
};
