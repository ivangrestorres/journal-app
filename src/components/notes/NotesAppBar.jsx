import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startSaveNote, startUploading } from "../../actions/notes";

export const NotesAppBar = () => {
    const dispatch = useDispatch();
    const { active: note } = useSelector((state) => state.notes);

    const handleSaveNote = () => {
        dispatch(startSaveNote(note));
    };

    const handleFileChange = ({ target }) => {
        const file = target.files[0];
        if (file) {
            dispatch(startUploading(file));
        }
    };

    const handlePicture = () => {
        document.querySelector("#fileSelector").click();
    };

    return (
        <div className={"notes_appbar"}>
            <span>28 de agosto de 2020</span>
            <input
                id={"fileSelector"}
                onChange={handleFileChange}
                type="file"
                style={{ display: "none" }}
            />
            <div>
                <button onClick={handlePicture} className={"btn"}>
                    Picture
                </button>
                <button onClick={handleSaveNote} className={"btn"}>
                    Save
                </button>
            </div>
        </div>
    );
};
