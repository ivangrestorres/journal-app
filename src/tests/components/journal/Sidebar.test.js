import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom";
import { Sidebar } from "../../../components/journal/Sidebar";
import { startLogout } from "../../../actions/auth";
import { addNote } from "../../../actions/notes";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock("../../../actions/auth", () => ({
    startLogout: jest.fn(),
}));

jest.mock("../../../actions/notes", () => ({
    addNote: jest.fn(),
}));

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null,
    },
    notes: {
        active: {},
        notes: [],
    },
};
let store = mockStore(initState);
store.dispatch = jest.fn();

describe("Pruebas en <Sidebar />", () => {
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    const wrapper = mount(
        <Provider store={store}>
            <Sidebar></Sidebar>
        </Provider>
    );

    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de llamar al logout", () => {
        wrapper.find("button").prop("onClick")();
        expect(startLogout).toHaveBeenCalled();
    });

    test("debe de llamar el startNewNote", () => {
        wrapper.find(".journal_new-entry").prop("onClick")();
        expect(addNote).toHaveBeenCalled();
    });
});
