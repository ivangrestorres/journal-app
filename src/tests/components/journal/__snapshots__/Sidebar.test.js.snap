// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Sidebar /> debe de mostrarse correctamente 1`] = `
<aside
  className="journal_sidebar"
>
  <div
    className="journal_sidebar-navbar"
  >
    <h4
      className="mt-1"
    >
      <i
        className="far fa-user"
      />
      <span>
         
      </span>
    </h4>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal_new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-3x"
    />
    <p
      className="mt-5"
    >
      New entry
    </p>
  </div>
  <div
    className="journal_entries"
  />
</aside>
`;
